mne_bids/read.py

```diff
@@ -71,6 +71,105 @@ def _read_raw(raw_fname, electrode=None, hsp=None, hpi=None, config=None,
+
+    Handle onset, duration, and description of each event.
+    """
+    events_dict = _from_tsv(events_fname)
+
+    # Get the descriptions of the events
+    if 'trial_type' in events_dict:
+        # Drop events unrelated to a trial type
+        events_dict = _drop(events_dict, 'n/a', 'trial_type')
+        descriptions = np.asarray(events_dict['trial_type'], dtype=str)
+
+    # If we don't have a proper description of the events, perhaps we have
+    # at least an event value?
+    elif 'value' in events_dict:
+        # Drop events unrelated to value
+        events_dict = _drop(events_dict, 'n/a', 'value')
+        descriptions = np.asarray(events_dict['value'], dtype=str)
+
+    # Worst case, we go with 'n/a' for all events
+    else:
+        descriptions = 'n/a'
+
+    # Add Events to raw as annotations
+    onsets = np.asarray(events_dict['onset'], dtype=float)
+    durations = np.asarray(events_dict['duration'], dtype=float)
+    annot_from_events = mne.Annotations(onset=onsets,
+                                        duration=durations,
+                                        description=descriptions,
+                                        orig_time=raw.info['meas_date'])
+    raw.set_annotations(annot_from_events)
+
+    return raw
+
+
+def _handle_channels_reading(channels_fname, bids_fname, raw):
+    """Read associated channels.tsv and populate raw.
+
+    Updates status (bad) and types of channels.
+    """
+    channels_dict = _from_tsv(channels_fname)
+
+    # First, make sure that ordering of names in channels.tsv matches the
+    # ordering of names in the raw data. The "name" column is mandatory in BIDS
+    ch_names_raw = list(raw.ch_names)
+    ch_names_tsv = channels_dict['name']
+    if ch_names_raw != ch_names_tsv:
+
+        msg = ('Channels do not correspond between raw data and the '
+               'channels.tsv file. For MNE-BIDS, the channel names in the '
+               'tsv MUST be equal and in the same order as the channels in '
+               'the raw data.\n\n'
+               '{} channels in tsv file: "{}"\n\n --> {}\n\n'
+               '{} channels in raw file: "{}"\n\n --> {}\n\n'
+               .format(len(ch_names_tsv), channels_fname, ch_names_tsv,
+                       len(ch_names_raw), bids_fname, ch_names_raw)
+               )
+
+        # this could be due to MNE inserting a 'STI 014' channel as the last
+        # channel: In that case, we can work
+        if not (ch_names_raw[-1] == 'STI 014' and
+                ch_names_raw[:-1] == ch_names_tsv):
+            raise RuntimeError(msg)
+
+    # Now we can do some work.
+    # The "type" column is mandatory in BIDS. We can use it to set channel
+    # types in the raw data using a mapping between channel types
+    channel_type_dict = dict()
+
+    # Get the best mapping we currently have from BIDS to MNE nomenclature
+    bids_to_mne_ch_types = _get_ch_type_mapping(from_mne_to_bids=False)
```

Comment:
I find this argument a bit confusing. Can you call `from_mne_to_bids` something else? Maybe something like:

```py
_get_ch_type_mapping('mne', 'bids')
```

and:

```py
_get_ch_type_mapping('bids', 'mne')
```
?

Not sure ...

Reply:
I also like:

`_get_ch_type_mapping(from='mne', to='bids')`